### PR TITLE
Add Check, CheckCreate, CheckCash, CheckCancel

### PR DESCRIPTION
**Pull request by intelliot**
LEDGER_ENTRY_TYPES from:
https://github.com/ripple/rippled/blob/eaff9a0e6aec0ad077f118501791c7684debcfd5/src/ripple/protocol/LedgerFormats.h#L36

TRANSACTION_TYPES from:
https://github.com/ripple/rippled/blob/eaff9a0e6aec0ad077f118501791c7684debcfd5/src/ripple/protocol/TxFormats.h#L33

### CODE REVIEW

**Comment by sharafian** [comment]
Is there anywhere I can find docs on what "Checks" are?

**Comment by wilsonianb** [comment]
https://ripple.com/build/known-amendments/#checks

**Comment by intelliot** [comment]
Most of the diff is just alphabetization/reordering to match rippled's source code. Only Check, CheckCreate, CheckCash, CheckCancel are new.

**Comment by wilsonianb** [comment]
It looks like [`sfCheckID`](https://github.com/ripple/rippled/pull/2245/files#diff-822bd938f72f71e9b5dd661e361ad215R174) and [`tecEXPIRED`](https://github.com/ripple/rippled/pull/2245/files#diff-a893f554c7286106e9df4e487fac0b60R75) were also added to support checks.
Unrelated, [`sfConsensusHash`](https://github.com/ripple/rippled/commit/e0168b98d7fdf4cdb846f7648b597c9274de4032) is missing here as well.

File: src/enums/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+### isVLEncoded
+
+If true, the field is Variable Length encoded. The variable-length encoded fields are `STI_VL`/`Blob`, `STI_ACCOUNT`/`AccountID`, and `STI_VECTOR256`/`Vector256`.
+
+### isSerialized
+
+Fields are serialized if they are not [one of these](https://github.com/ripple/rippled/blob/eaff9a0e6aec0ad077f118501791c7684debcfd5/src/ripple/protocol/impl/SField.cpp#L71-L78) or if they are not an SField.
+
+- https://github.com/ripple/ripple-binary-codec/blob/master/src/enums/definitions.json#L832
```
Comment (wilsonianb) [inline src/enums/README.md]: Safer to link to a specific commit hash:
`https://github.com/ripple/ripple-binary-codec/blob/14e76e68ead7e4bcd83c942dbdc9064d5a66869b/src/enums/definitions.json#L832`

**Comment by sublimator** [comment]
You guys want this converted to TypeScript ?

**Pull request merged**